src/http_server.cc

```diff
@@ -3015,40 +3063,36 @@ HTTPAPIServer::ForwardHeaders(
 }
 
 void
-HTTPAPIServer::HandleInfer(
+HTTPAPIServer::HandleGenerate(
     evhtp_request_t* req, const std::string& model_name,
-    const std::string& model_version_str)
+    const std::string& model_version_str, bool streaming)
 {
+  AddContentTypeHeader(req, "application/json");
   if (req->method != htp_method_POST) {
     RETURN_AND_RESPOND_WITH_ERR(
         req, EVHTP_RES_METHNALLOWED, "Method Not Allowed");
   }
 
   int64_t requested_model_version;
   RETURN_AND_RESPOND_IF_ERR(
-      req, GetModelVersionFromString(
-               model_version_str.c_str(), &requested_model_version));
-  RETURN_AND_RESPOND_IF_ERR(
-      req, CheckTransactionPolicy(req, model_name, requested_model_version));
+      req,
+      GetModelVersionFromString(model_version_str, &requested_model_version));
 
   // If tracing is enabled see if this request should be traced.
   TRITONSERVER_InferenceTrace* triton_trace = nullptr;
   std::shared_ptr<TraceManager::Trace> trace =
       StartTrace(req, model_name, &triton_trace);
 
-  // Decompress request body if it is compressed in supported type
-  evbuffer* decompressed_buffer = nullptr;
-  RETURN_AND_RESPOND_IF_ERR(req, DecompressBuffer(req, &decompressed_buffer));
-
-  // Get content length as a default header_length if no header specified
-  int32_t content_length = 0;
+  std::map<std::string, triton::common::TritonJson::Value> input_metadata;
+  triton::common::TritonJson::Value meta_data_root;
   RETURN_AND_RESPOND_IF_ERR(
-      req, GetContentLength(req, decompressed_buffer, &content_length));
+      req, ModelInputMetadata(
+               model_name, requested_model_version, &input_metadata,
+               &meta_data_root));
 
-  // Get the header length
-  size_t header_length = 0;
-  RETURN_AND_RESPOND_IF_ERR(
-      req, GetInferenceHeaderLength(req, content_length, &header_length));
+
+  // [FIXME] decompression should have been done here. before parsing request
```

Comment:
We should keep in mind to document these limitations. However, I don't think compression is any useful for text data..